<launch>
    <rosparam file="$(find kantonatsu)/param/logitech_f710_joy_assign.yaml" command="load" />

    <arg name="use_joy" value="true" />
    <!-- base controller -->
    <node pkg="base_controller" type="base_controller" name="base_ctrl" output="screen">
        <param name="motor_max_acc" value="50.0" />
        <param name="motor_max_vel" value="80.0" />
        <param name="invert_x" value="false" />
        <param name="invert_y" value="false" />
        <param name="invert_z" value="false" />
        <remap from="motor0_cmd_vel" to="base/motor0_cmd_vel" />
        <remap from="motor1_cmd_vel" to="base/motor1_cmd_vel" />
        <remap from="motor2_cmd_vel" to="base/motor2_cmd_vel" />
    </node>
    
    <!-- joy node -->
    <node respawn="true" pkg="joy" type="joy_node" name="joy_node" >
        <param name="dev" type="string" value="/dev/input/js0" />
        <param name="deadzone" value="0.12" />
    </node>

    <!--  navigation -->
    
    
    <node pkg="kantoharu20_3wheel" type="haru20_b_joy_main" name="main" output="screen" />
    <!-- shutdown controller -->
    <node pkg="base_controller" type="shutdown_controller_joy" name="shutdown_ctrl" />
    
    <node pkg="kantonatsu_3wheel" type="natu19_b_mr1_can" name="mr1_can" />
    
    <!-- usb_can -->
    <node pkg="kantoharu20_3wheel" type="usb_can_node_19" name="usb_can_node_19" respawn="true" output="screen">
        <param name="port" value="/dev/ttyUSB0" />
    </node>

    <!-- odometry broadcaster -->
    <node pkg="base_controller" type="odom_publisher" name="odom_publisher" if="true">
        <param name="ctrl_freq" value="120" />
        <param name="odom_frame" value="odom" />
        <remap from="/odom/yaw" to="/base/odom/yaw" />
        <remap from="/odom/x" to="/base/odom/x" />
        <remap from="/odom/y" to="/base/odom/y" />
    </node>
    
    
</launch>
